aws_lambda_powertools/metrics/metrics.py

```diff
@@ -82,7 +82,12 @@ def clear_metrics(self):
         self.metric_set.clear()
         self.dimension_set.clear()
 
-    def log_metrics(self, lambda_handler: Callable[[Any, Any], Any] = None, capture_cold_start_metric: bool = False):
+    def log_metrics(
+        self,
+        lambda_handler: Callable[[Any, Any], Any] = None,
+        capture_cold_start_metric: bool = False,
+        raise_on_empty_metrics: bool = False,
```

Comment:
Shouldn't this be inverted?

Looking at the changes, the current behaviour is to raise an exception on empty metrics, so this would make a change to silently fail on empty metrics. From a developer perspective, I think it'd be better to fail first as it might be due to an error in the code that doesn't trigger a metric under certain conditions. Then a developer can choose to either ignore (set this to False) or fix the issue if there was one.

Reply:
Discussed offline, agreed on warning when metric set is empty.